[doc][yba] Patching nodes and upgrade notes (#21099)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/manage-deployments/upgrade-software.md

- On-premises provider - if you use on-premises providers with manually-provisioned nodes, you will need to update your current procedures for manually provisioning instances, to include installing node agent. Refer to [Manually provision on-premises nodes](../configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual/#install-node-agent).

- Upgrade OS - for universes created using an on-premises provider with manually-provisioned nodes, if your OS patching procedures involve re-installing YugabyteDB software on a node, you will need to update those procedures to accommodate node agent.

- Firewall ports - update your firewall rules to allow incoming TCP traffic on port 9000 for all nodes in a universe.

## View and import YugabyteDB releases into YugabyteDB Anywhere

Before you can upgrade your universe to a specific version of YugabyteDB, verify that the release is available and, if necessary, import the release.

To view the releases that are available, do the following:

- Click the user profile icon and choose **Releases**.

    ![Releases](/images/yp/releases-list.png)

If a release that you want to install on a universe is not available, import it as follows:

> 1. On the **Releases** page, click **Import** to open the **Import Releases** dialog shown in the following illustration:

Reviewer: ```suggestion
1. On the **Releases** page, click **Import** to open the **Import Releases** dialog as shown in the following illustration:
```

The YugabyteDB release that is powering a universe can be upgraded to get new features and fixes included in the release.

## Before you begin

> You cannot upgrade a universe to a version of YugabyteDB that is later than the version of YugabyteDB Anywhere. To upgrade to a more recent version of YugabyteDB, you may first have to [upgrade YugabyteDB Anywhere](../../upgrade/).

Reviewer: ```suggestion
You cannot upgrade a universe to a version of YugabyteDB that is later than the version of YugabyteDB Anywhere (YBA). So, to upgrade to a more recent version of YugabyteDB, you may first have to [upgrade YugabyteDB Anywhere](../../upgrade/).
```

In addition, before starting the upgrade, review the following major changes in previous YugabyteDB releases. Depending on the upgrade you are planning, you may need to make changes to your automation.

Changes in these releases are cumulative. For example, to upgrade from v2.14 to v2.20, your automation will need to take into account the changes in v2.16 and v2.18.

### Upgrading from versions earlier than v2.16.0

> The YB Controller (YBC) service was introduced in v2.16.0 for all universes (except Kubernetes), and is required for YugabyteDB Anywhere 2.16.0 and later.

Reviewer: ```suggestion
The YB Controller (YBC) service was introduced in v2.16.0 for all universes (except Kubernetes), and is required for YBA 2.16.0 and later.
```

  - Setting systemd-specific database service unit files (if used), and configuring TCP ports. Refer to [Manually provision on-premises nodes](../../configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual/).
  - After upgrading nodes, manually installing YBC on the nodes. Refer to [Upgrade manually-provisioned on-premises universe](#upgrade-manually-provisioned-on-premises-universe).

- Upgrade OS - for universes created using an on-premises provider with manually-provisioned nodes, if your OS patching procedures involve re-installing YugabyteDB software on a node, you will need to update those procedures to accommodate YBC.

- Firewall ports - update your firewall rules to allow incoming TCP traffic on port 18018, which is used by YBC, for all nodes in a universe.

### Upgrading from versions earlier than v2.18.0

YBC was introduced for Kubernetes clusters in v2.18.0. Refer to [Upgrading to 2.16.0](/#upgrading-to-v2-16-0).

### Upgrading from versions earlier than v2.18.2

> The Node Agent was introduced for all universes in v2.18.2. Node agent is an RPC service running on a YugabyteDB node, and is used to manage communication between YugabyteDB Anywhere and the nodes in universes. Except for Day 0 tasks during initial installation, YBA no longer uses SSH and SCP to manage nodes; instead, YBA connects to the Node agent process listening on port 9000, and performs all its management via this secure connection. For more information, refer to the [Node agent FAQ](../../../faq/yugabyte-platform/#node-agent).

Reviewer: ```suggestion
The Node Agent was introduced for all universes in v2.18.2. Node agent is an RPC service running on a YugabyteDB node, and is used to manage communication between YBA and the nodes in universes. Except for Day 0 tasks during initial installation, YBA no longer uses SSH and SCP to manage nodes; instead, YBA connects to the Node agent process listening on port 9000, and performs all its management via this secure connection. For more information, refer to the [Node agent FAQ](../../../faq/yugabyte-platform/#node-agent).
```
